includes/Core/Remote_Features/Remote_Features_Fallback.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Remote_Features\Remote_Features_Fallback
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Remote_Features;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Credentials;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Util\BC_Functions;
+
+/**
+ * Class providing the integration of remote features.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Remote_Features_Fallback {
+
+	/**
+	 * Context instance.
+	 *
+	 * @var Context
+	 */
+	private Context $context;
+
+	/**
+	 * Credentials instance.
+	 *
+	 * @var Credentials
+	 */
+	private Credentials $credentials;
+
+	/**
+	 * Remote_Features instance.
+	 *
+	 * @var Remote_Features
+	 */
+	private Remote_Features $setting;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context     $context     Context instance.
+	 * @param Options     $options     Options instance.
+	 * @param Credentials $credentials Credentials instance.
+	 */
+	public function __construct(
+		Context $context,
+		Options $options,
+		Credentials $credentials
+	) {
+		$this->context     = $context;
+		$this->credentials = $credentials;
+		$this->setting     = new Remote_Features( $options );
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_action( 'wp_ajax_sk_pull_remote_features_fallback', array( $this, 'pull_remote_features_ajax_fallback' ) );
+	}
+
+	/**
+	 * Calls the CRON action.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function pull_remote_features_ajax_fallback() {
+		check_ajax_referer( 'remote_features_fallback' );
+
+		do_action( Remote_Features_Cron::CRON_ACTION );
+		wp_send_json_success();
+	}
+
+	/**
+	 * Fallback for syncing the remote features if CRON did not run.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function remote_features_sync_fallback() {
+		if ( $this->credentials->has() && ! $this->credentials->using_proxy() ) {
+			return;
+		}
+
+		// Ensure we run the fallback only if user is on the Site Kit dashboard.
+		if ( 'googlesitekit-dashboard' !== $this->context->input()->filter( INPUT_GET, 'page' ) ) {
+			return;
+		}
```

Comment:
We shouldn't rely on query parameters to determine the page we're on. This is already handled by core, so we can leverage the current screen or only hook the method on to a screen-specific hook. Our screen class exposes a function we can define that is called on load https://github.com/google/site-kit-wp/blob/64b8643c162ffff4fc877a51f2fb43c0ba4f45c8/includes/Core/Admin/Screens.php#L233-L238

Reply:
Since we changed approach to ride along the heartbeat tick, this hook won't be used. Thanks for pointing this out, haven't seen this before.